src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,625 @@
+        return iamConfiguration;
+    }
+
+    public void setIamConfiguration(BucketIamConfiguration iamConfiguration) {
+        this.iamConfiguration = iamConfiguration;
+    }
+
+    /**
+     * The bucket's lifecycle configuration. See also `Object Lifecycle Management <https://cloud.google.com/storage/docs/lifecycle/>`_.
+     *
+     * @subresource gyro.google.storage.BucketLifecycle
+     */
+    @Updatable
+    public BucketLifecycle getLifecycle() {
+        return lifecycle;
+    }
+
+    public void setLifecycle(BucketLifecycle lifecycle) {
+        this.lifecycle = lifecycle;
+    }
+
+    /**
+     * The bucket's logging configuration.
+     *
+     *  @subresource gyro.google.storage.BucketLogging
+     */
+    @Updatable
+    public BucketLogging getLogging() {
+        return logging;
+    }
+
+    public void setLogging(BucketLogging logging) {
+        this.logging = logging;
+    }
+
+    /**
+     * Minimum age an object in the bucket must reach before it can be deleted or overwritten. See also `Retention policies <https://cloud.google.com/storage/docs/bucket-lock?#retention-policy/>`_.
+     *
+     * @subresource gyro.google.storage.BucketRetentionPolicy
+     */
+    @Updatable
+    public BucketRetentionPolicy getRetentionPolicy() {
+        return retentionPolicy;
+    }
+
+    public void setRetentionPolicy(BucketRetentionPolicy retentionPolicy) {
+        this.retentionPolicy = retentionPolicy;
+    }
+
+    /**
+     * Bucket's default storage class used whenever no ``storageClass`` is specified for a newly-created object. Valid values are ``STANDARD``, ``NEARLINE``, ``COLDLINE``, ``MULTI-REGIONAL``, ``REGIONAL`` or ``DURABLE_REDUCED_AVAILABILITY``. Defaults to ``STANDARD``.
+     */
+    @Updatable
+    @ValidStrings({"STANDARD", "NEARLINE", "COLDLINE", "MULTI-REGIONAL", "REGIONAL", "DURABLE_REDUCED_AVAILABILITY"})
+    public String getStorageClass() {
+        return storageClass != null ? storageClass.toUpperCase() : null;
+    }
+
+    public void setStorageClass(String storageClass) {
+        this.storageClass = storageClass;
+    }
+
+    /**
+     * The bucket's versioning configuration.
+     *
+     * @subresource gyro.google.storage. BucketVersioning
+     */
+    @Updatable
+    public BucketVersioning getVersioning() {
+        return versioning;
+    }
+
+    public void setVersioning(BucketVersioning versioning) {
+        this.versioning = versioning;
+    }
+
+    /**
+     * The bucket's website configuration controlling how the service behaves when accessing bucket contents as a web site.
+     *
+     * @subresource gyro.google.storage.BucketWebsite
+     */
+    @Updatable
+    public BucketWebsite getWebsite() {
+        return website;
+    }
+
+    public void setWebsite(BucketWebsite website) {
+        this.website = website;
+    }
+
+    /**
+     * The generated URI of this bucket. (Read Only)
+     */
+    public String getSelfLink() {
```

Comment:
Should have the annotation `@Output`
And `(Read Only)` from the doc should be removed, as having the `@Output` annotation will move this to a separate section for output variables in the generated docs.

Reply:
Done.